### PR TITLE
[Feature-7024] Add waiting strategy to support master/worker can recover from registry lost

### LINKED ISSUE #7024: [Improvement][MasterWorker] Self-recovery when master or worker lost connection from registry center
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Background

Now if a master or worker lost zk connection, it doesn't disconver itself immediately until checking the dead server list.
And when it knows that it was judged to dead serve, it will stop itself, without self-recovery.

### Prosoal

d264093da134508c29b021d9c57d440

When master lost zk connection:
1. update current server state to `wait reconnect`
2. send server lost connection alert
3. keep quartz working (it will ensure work normally by quartz and db)
4. stop accepting new request
4. stop handling commands and process instances, clear the local running process instances; (it will be take over by other master)
5. wait to reconnect
6. when reconnect successfully, send server recover alert, update server state to `normal` and recover working

628345e6b1dd3cf147f8d40ee35021f

When worker lost zk connection:
1. update current server state to `wait reconnect`
2. send server lost connection alert
2. kill the running task (it will be task over by master and rerun)
3. stop accepting new request
4. wait to reconnect within a certain time
5. if reconnect timeout, stop itself
6. when reconnect successfully,  send server recover alert, update server state to `normal` and recover working;

_Originally posted by @caishunfeng in https://github.com/apache/dolphinscheduler/discussions/6643#discussioncomment-1706255_

### Related issues

#7004 

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Purpose of the pull request

Close #7024 

## Brief change log
- Add ConnectStrategy used for registry connection disconnect/reconnect
- Add stop/waiting strategy
- Use ServerLifyCycleManager to replace Stopper

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/cron/CronUtils.java [hunk 1]
```diff
@@ -147,7 +143,7 @@ public static List<ZonedDateTime> getFireDateList(@NonNull ZonedDateTime startTi
         List<ZonedDateTime> dateList = new ArrayList<>();
         ExecutionTime executionTime = ExecutionTime.forCron(cron);
 
-        while (Stopper.isRunning()) {
+        while (!ServerLifeCycleManager.isStopped()) {
```
Comment (caishunfeng) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/cron/CronUtils.java:146]: Should we check server status here? Maybe it's no need.

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/registry/RegistryClient.java [hunk 2]
```diff
@@ -142,33 +139,8 @@ public Map<String, String> getServerMaps(NodeType nodeType, boolean hostOnly) {
 
     public boolean checkNodeExists(String host, NodeType nodeType) {
         return getServerMaps(nodeType, true).keySet()
-                                            .stream()
-                                            .anyMatch(it -> it.contains(host));
-    }
-
-    public void handleDeadServer(Collection<String> nodes, NodeType nodeType, String opType) {
```
Comment (caishunfeng) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/registry/RegistryClient.java:149]: It's necessary to update the docs.

File: dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/runner/WorkerManagerThread.java [hunk 3]
```diff
@@ -153,8 +154,14 @@ public void run() {
...
                 logger.error("An unexpected interrupt is happened, "
-                    + "the exception will be ignored and this thread will continue to run", e);
+                        + "the exception will be ignored and this thread will continue to run", e);
             }
         }
     }
+
+    public void clearTask() {
+        waitSubmitQueue.clear();
+        workerExecService.getTaskExecuteThreadMap().values().forEach(TaskExecuteThread::kill);
```
Comment (caishunfeng) [inline dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/runner/WorkerManagerThread.java:164]: Is it better to add try catch for kill action?

**Review by caishunfeng — COMMENTED** [review]
@ruanwenjun Please add/update the docs.

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/cron/CronUtils.java (hunk 1 above)
Comment (ruanwenjun) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/cron/CronUtils.java:146]: Yes, we don't need to check the server status in all places.

File: dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/runner/WorkerManagerThread.java (hunk 3 above)
Comment (ruanwenjun) [reply dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/runner/WorkerManagerThread.java:164]: The kill action is safe, will not throw any exception.

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/registry/RegistryClient.java (hunk 2 above)
Comment (ruanwenjun) [reply dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/registry/RegistryClient.java:149]: Done

**Comment by ruanwenjun** [comment]
@zhongjiajie This PR has some changes about the doc, please take a look.

**Review by caishunfeng — APPROVED** [review]
LGTM

**Pull request merged**